tests/js/dataMock/analytics.js

```diff
@@ -0,0 +1,311 @@
+	// dimension set in the combined stream (array). We need to use array of streams because report arguments may
+	// have 0 or N dimensions (N > 1) which means that in the each row of the report data we will have an array
+	// of dimension values.
+	const dimensions = Array.isArray( args.dimensions ) ? args.dimensions : [ args.dimensions ];
+	dimensions.forEach( ( dimension ) => {
+		if ( dimension === 'ga:date' ) {
+			// Generates a stream (an array) of dates when the dimension is ga:date.
+			streams.push( new Observable( ( observer ) => {
+				const currentDate = new Date( args.startDate );
+				const end = new Date( args.endDate );
+
+				while ( currentDate.getTime() <= end.getTime() ) {
+					observer.next( currentDate.toISOString().split( 'T' )[ 0 ].replace( /\D/g, '' ) );
+					currentDate.setDate( currentDate.getDate() + 1 );
+				}
+
+				observer.complete();
+			} ) );
+		} else if ( dimension && typeof ANALYTICS_DIMENSION_OPTIONS?.[ dimension ] === 'function' ) {
+			// Generates a stream (an array) of dimension values using a function associated with the current dimension.
+			streams.push( new Observable( ( observer ) => {
+				for ( let i = 1; i <= 90; i++ ) { // 90 is the max number of dates in the longest date range.
+					const val = ANALYTICS_DIMENSION_OPTIONS[ dimension ]( i );
+					if ( val ) {
+						observer.next( val );
+					} else {
+						break;
+					}
+				}
+
+				observer.complete();
+			} ) );
+		} else if ( dimension && Array.isArray( ANALYTICS_DIMENSION_OPTIONS?.[ dimension ] ) ) {
+			// Uses predefined array of dimension values to create a stream (an array) from.
+			streams.push( from( ANALYTICS_DIMENSION_OPTIONS[ dimension ] ) );
+		} else {
+			// In case when a dimension is not provided or is not recognized, we use NULL to create a stream (an array) with just one value.
+			streams.push( from( [ null ] ) );
+		}
+	} );
+
+	// This is the list of operations that we apply to the cobmined stream (array) of dimension values.
+	const ops = [
+		// Convert a dimension value to a row object and generate metric values.
+		map( ( dimensionValue ) => ( {
+			dimensions: Array.isArray( dimensionValue ) ? dimensionValue : [ dimensionValue ],
+			metrics: generateMetricValues( getMetricType( validMetrics[ 0 ] ), metricValuesCount ),
+		} ) ),
+		// Make sure we take the appropriate number of rows.
+		take( args.limit > 0 ? +args.limit : 90 ),
+		// Accumulate all rows into a single array.
+		reduce( ( rows, row ) => [ ...rows, row ], [] ),
+		// Sort rows if args.orderby is provided.
+		map( ( rows ) => args.orderby ? sortRows( rows, validMetrics, args.orderby ) : rows ),
+	];
+
+	// Process the stream of dimension values and add generated rows to the report data object.
+	zip( ...streams ).pipe( ...ops ).subscribe( ( rows ) => {
+		data.rows = rows;
+		data.rowCount = rows.length;
+
+		data.minimums = [ ...( rows[ 0 ]?.metrics || [] ) ];
+		data.maximums = [ ...( rows[ rows.length - 1 ]?.metrics || [] ) ];
```

Comment:
I don't think these should ever be empty arrays. These should be the actual minimum and maximum values present in the rows for the respective value column. Also, there may be multiple entries (`metricValuesCount`).

Reply:
I didn't want to overcomplicate the logic here because we don't need mathematically correct values anyway. Fallback values have been added just to be extra safe here.

Reply:
In cases like this, where the implementation may be inconsistent with what would be expected, let's make sure to document it as an intentional choice to make it clear that this isn't a bug.

Reply:
Added comments.